app/parse.py

```diff
@@ -8,8 +16,48 @@ class Quote:
     tags: list[str]
 
 
+def parse_single_quote(quote_soup: BeautifulSoup) -> Quote:
+    return Quote(
+        text=quote_soup.select_one(".text").text,
+        author=quote_soup.select_one(".author").text,
+        tags=[tag_soup.text for tag_soup in quote_soup.select(".tag")],
+    )
+
+
+def get_single_page_quotes(page_soup: BeautifulSoup) -> list[Quote]:
+    quotes = page_soup.select(".quote")
+    return [parse_single_quote(quote) for quote in quotes]
+
+
+def get_all_quotes() -> list[Quote]:
+    i = 1
+    quotes = []
+    while True:
+        page_url = urljoin(BASE_URL, f"/page/{i}")
```

Comment:
The urljoin(BASE_URL, f"/page/{i}") is used to form the page URL, but this could result in an incorrect URL because the leading slash (/) in f"/page/{i}" would cause urljoin to ignore the path part of BASE_URL and append the page path directly to the domain. It should be f"page/{i}" instead.